api/stream/design.go

```diff
@@ -44,12 +44,18 @@ func init() {
 		Package: "github.com/keboola/keboola-as-code/internal/pkg/service/stream/dependencies",
 		DependenciesTypeFn: func(method *service.MethodData) string {
 			if dependencies.HasSecurityScheme("APIKey", method) {
+				if strings.Contains(method.PayloadDef, "\tBranchID") {
+					return "dependencies.BranchRequestScope"
+				}
 				return "dependencies.ProjectRequestScope"
```

Comment:
Endpoints with the `BranchID` in payload should use the new `BranchRequestScope`.